storage/client/queue/src/main/java/com/azure/storage/queue/QueueAsyncClient.java

```diff
@@ -1,29 +1,479 @@
+import com.azure.storage.queue.models.StorageErrorException;
+import com.azure.storage.queue.models.UpdatedMessage;
+import java.net.MalformedURLException;
 import java.net.URL;
+import java.time.Duration;
+import java.util.List;
+import java.util.Map;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+/**
+ * This class provides a client that contains all the operations for interacting with a queue in Azure Storage Queue.
+ * Operations allowed by the client are creating and deleting the queue, retrieving and updating metadata and access
+ * policies of the queue, and enqueuing, dequeuing, peeking, updating, and deleting messages.
+ *
+ * <p><strong>Instantiating an Asynchronous Queue Client</strong></p>
+ *
+ * <pre>
+ * QueueAsyncClient client = QueueAsyncClient.builder()
+ *     .connectionString(connectionString)
+ *     .endpoint(endpoint)
+ *     .queueName(queueName)
+ *     .buildAsyncClient();
+ * </pre>
+ *
+ * <p>View {@link QueueClientBuilder this} for additional ways to construct the client.</p>
+ *
+ * @see QueueClientBuilder
+ * @see QueueClient
+ * @see SharedKeyCredential
+ * @see SASTokenCredential
+ */
+public final class QueueAsyncClient {
+    private final AzureQueueStorageImpl client;
+    private final String queueName;
+
+    /**
+     * Creates a QueueAsyncClient that sends requests to the storage queue service at {@link AzureQueueStorageImpl#url() endpoint}.
+     * Each service call goes through the {@link HttpPipeline pipeline} in the {@code client}.
+     *
+     * @param client Client that interacts with the service interfaces
+     * @param queueName Name of the queue
+     */
+    QueueAsyncClient(AzureQueueStorageImpl client, String queueName) {
+        this.queueName = queueName;
+
+        this.client = new AzureQueueStorageBuilder().pipeline(client.httpPipeline())
+            .url(client.url())
+            .version(client.version())
+            .build();
+    }
+
+    /**
+     * Creates a QueueAsyncClient that sends requests to the storage queue service at {@code endpoint}.
+     * Each service call goes through the {@code httpPipeline}.
+     *
+     * @param endpoint URL for the Storage Queue service
+     * @param httpPipeline HttpPipeline that the HTTP requests and response flow through
+     * @param queueName Name of the queue
+     */
+    QueueAsyncClient(URL endpoint, HttpPipeline httpPipeline, String queueName) {
+        this.queueName = queueName;
+
+        this.client = new AzureQueueStorageBuilder().pipeline(httpPipeline)
+            .url(endpoint.toString())
+            .build();
+    }
+
+    /**
+     * @return the URL of the storage queue
+     * @throws RuntimeException If the queue is using a malformed URL.
+     */
+    public URL getQueueUrl() {
+        try {
+            return new URL(client.url());
+        } catch (MalformedURLException ex) {
+            throw new RuntimeException("Queue URL is malformed");
```

Comment:
ClientLogger.logAndThrow all exceptions. Please search for all cases because soon Shawn will check in his CheckStyle rule and it will break the build.

Reply:
Do you mean I added a log message before throw statement. Done.